imap/xapian_wrap.cpp

```diff
@@ -518,20 +513,57 @@ int xapian_compact_dbs(const char *dest, const char **sources)
             }
         }
         thispath = "(unknown path)";
+        int flags = Xapian::DB_DANGEROUS|Xapian::DB_NO_SYNC|Xapian::DB_CREATE|Xapian::DB_BACKEND_GLASS;
+        std::string metadir = std::string(dest) + ".META";
+        Xapian::WritableDatabase metadb {metadir, flags};
 
-        // Compact database.
-        static CyrusMetadataCompactor comp;
-        // FULLER because we never write to compression targets again.
-        db.compact(dest, Xapian::Compactor::FULLER | Xapian::DBCOMPACT_MULTIPASS, 0, comp);
+        // set version metadata
+        std::ostringstream val;
+        for (std::set<int>::iterator it = db_versions.begin(); it != db_versions.end(); ++it) {
+            if (it != db_versions.begin()) val << ",";
+            val << *it;
+        }
+        if (db.get_metadata("cyrus.db_version").empty()) {
+            metadb.set_metadata("cyrus.db_version", val.str());
+        }
+        else {
+            metadb.set_metadata("cyrus.db_version", std::string("OVERWRITE:") + val.str());
+        }
+        if (db.get_metadata("cyrus.stem-version").empty()) {
```

Comment:
Yeah, that's not actually necessary - it's a noop, setting a metadata value to blank is the same as removing it, but since this is a new DB clearly it's not going to have that value!  The other path ("OVERWRITE:") is necessary, because that will cause it to replace any metadata value from the other databases in what gets written to the new one :)